common/azure-common/src/main/java/com/azure/common/implementation/SwaggerMethodParser.java

```diff
@@ -267,6 +276,37 @@ public String path(Object[] methodArguments) {
         return result;
     }
 
+    /**
+     * Get the encoded form parameters that have been added to this value based on the provided
+     * method arguments.
+     *
+     * @param swaggerMethodArguments the arguments that will be used to create the form parameters'
+     *                               values
+     * @return an Iterable with the encoded form parameters
+     */
+    public Iterable<EncodedParameter> encodedFormParameters(Object[] swaggerMethodArguments) {
+        final List<EncodedParameter> result = new ArrayList<>();
+        if (formSubstitutions != null) {
```

Comment:
My guess is that the common case will not have form parameters. If this is true, then I suggest inverting the null check (to `if (formSubmmissions == null) return Collections.emptyList()`), instead of creating a new arraylist every time.